Gems/Atom/Tools/AtomToolsFramework/Code/Include/AtomToolsFramework/Viewport/ModularViewportCameraController.h

```diff
@@ -141,21 +141,21 @@ namespace AtomToolsFramework
         AzFramework::Camera m_camera; //!< The current camera state (pitch/yaw/position/look-distance).
         AzFramework::Camera m_targetCamera; //!< The target (next) camera state that m_camera is catching up to.
         AzFramework::Camera m_previousCamera; //!< The state of the camera from the previous frame.
-        AZStd::optional<AzFramework::Camera> m_storedCamera; //!< A potentially stored camera for when a custom reference frame is set.
+        AZStd::optional<AzFramework::Camera> m_storedCamera; //!< A potentially stored camera for when a transform is being tracked.
         AzFramework::CameraSystem m_cameraSystem; //!< The camera system responsible for managing all CameraInputs.
         AzFramework::CameraProps m_cameraProps; //!< Camera properties to control rotate and translate smoothness.
         CameraControllerPriorityFn m_priorityFn; //!< Controls at what priority the camera controller should respond to events.
 
         CameraAnimation m_cameraAnimation; //!< Camera animation state (used during CameraMode::Animation).
         CameraMode m_cameraMode = CameraMode::Control; //!< The current mode the camera is operating in.
-        //! An additional reference frame the camera can operate in (identity has no effect).
-        AZ::Transform m_referenceFrameOverride = AZ::Transform::CreateIdentity();
-        //! Flag to prevent circular updates of the camera transform (while the viewport transform is being updated internally).
-        bool m_updatingTransformInternally = false;
+        float m_roll = 0.0f; //!< The current amount of roll to be applied to the camera.
```

Comment:
Out of curiosity, why not add roll to `AzFramework::Camera` instead? Its omission always struck me, though I imagine a fair amount of touchpoints would need to be updated.

Reply:
The reason is you rarely if ever want it for a normal Editor camera. Using roll makes the camera behave strangely, it's only really useful maybe in a flying game, even then it's debatable 😛 . This is just to ensure when tracking an Editor transform the camera remains faithful to the orientation. It _might_ be something we update in future but I'd rather hold off until it's needed otherwise it just complicates the code for something that may never be required.